runtime/compiler/ilgen/Walker.cpp

```diff
@@ -3352,6 +3379,48 @@ TR_J9ByteCodeIlGenerator::genInvokeHandle(TR::SymbolReference *invokeExactSymRef
    return callNode;
    }
 
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+void
+TR_J9ByteCodeIlGenerator::loadFromSideTableForInvokeDynamic(int32_t callSiteIndex)
+   {
+   TR::SymbolReference *appendixSymRef = symRefTab()->findOrCreateCallSiteTableEntrySymbol(_methodSymbol, callSiteIndex);
+   TR::Node * appendixNode = loadSymbol(TR::aload, appendixSymRef);
+   if (!appendixSymRef->isUnresolved())
+      appendixNode->setIsNonNull(true);
+   else
+      {
+      // Appendix being unresolved means the target is also unresolved.
+      // Instead of creating a call to the adapter method, we construct
+      // a call to linkToStatic and provide appendix and target as
+      // additional parameters.
+      TR::SymbolReference *memberNameSymRef = symRefTab()->findOrCreateCallSiteTableEntrySymbol(_methodSymbol, callSiteIndex, true);
+      TR::Node * memberNameNode = loadSymbol(TR::aload, memberNameSymRef);
+      appendixNode->setIsNull(true);
```

Comment:
Setting null flag on a node means that the node produce null value at runtime, which is wrong.